internal/resources/gitrepository/spec/spec.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package spec
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	gitrepositoryclustermodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/gitrepository/cluster"
+)
+
+var SpecSchema = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "Spec for the Repository.",
+	Required:    true,
+	MaxItems:    1,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			URLKey: {
+				Type:         schema.TypeString,
+				Description:  "URL of the git repository. Repository URL should begin with http, https, or ssh",
+				Required:     true,
+				ValidateFunc: validation.IsURLWithScheme([]string{"http", "https", "ssh"}),
```

Comment:
I am not is this is a valid input with http .. check with the team

Reply:
The API spec does allow `http` scheme and the same is also present in the description of the swagger-generated model which is directly picked up from the proto definition for this RPC.